### DISCUSSION TITLE
Tools for creating custom wizards

### DISCUSSION THREAD

**Discussion by computerquip-work** [wixtoolset/issues#7891] [Questions] [answered]
I've tried my hand at making a custom UI by hand. I really, really hated this whole process. It ended up being several hundred lines of XML that ended up being more complicated than what I think I'd find in say a procedural language doing the same thing.

Are there tools for UI creation? If not, do people really create their custom UIs by hand? I didn't find this that productive and extremely time consuming.

**Comment by chrpai** [comment, marked as answer]
IMO, less is more.  Most users just click next next next and don't read anything.  MSIX doesn't allow any custom UI at all.  Apps on iPhones and such don't either.  Whenever possible I minimize the UI as much as possible and defer it to first run.

My IsWiX tooling has a project template that uses the UI extension and includes an internal custom dialog.  To enable it you just uncomment one line of XML.  If you don't need it, just delete the wxs fragment.  

If you need a boostrapper, that's an option for UI also. In that scenario you just run your package(s) silently.

If you really need a lot of custom UI work in a stand alone MSI then you might want to consider InstallShield.  It has a very nice WYSIWYG dialog editor built into it's IDE.

**Reply by computerquip-work** [reply]
I think these are fair points. I'll reconsider how I make future installers...
